app/api/chemotion/admin_api.rb

```diff
@@ -290,22 +103,38 @@ class AdminAPI < Grape::API
             when 'NodeAdd'
               obj = Group.find(params[:id]) if %w[Group].include?(params[:rootType])
               obj = Device.find(params[:id]) if %w[Device].include?(params[:rootType])
-              new_users = (params[:add_users] || []).map(&:to_i) - obj.users.pluck(:id) if %w[Person Group].include?(params[:actionType])
-              new_users = (params[:add_users] || []).map(&:to_i) - obj.devices.pluck(:id) if %w[Device].include?(params[:actionType])
+              if %w[Person Group].include?(params[:actionType])
+                new_users = (params[:add_users] || []).map(&:to_i) - obj.users.pluck(:id)
+              end
+              if %w[Device].include?(params[:actionType])
+                new_users = (params[:add_users] || []).map(&:to_i) - obj.devices.pluck(:id)
+              end
               obj.users << Person.where(id: new_users) if %w[Person].include?(params[:actionType])
               obj.users << Group.where(id: new_users) if %w[Group].include?(params[:actionType])
               obj.devices << Device.where(id: new_users) if %w[Device].include?(params[:actionType])
               obj.save!
-              present obj, with: Entities::GroupDeviceEntity, root: 'root'
+
+              if obj.is_a?(Device)
+                present obj, with: Entities::DeviceEntity, root: 'root'
+              else
+                present obj, with: Entities::GroupDeviceEntity, root: 'root'
+              end
             when 'NodeDel'
               obj = Group.find(params[:id]) if %w[Group].include?(params[:rootType])
               obj = Device.find(params[:id]) if %w[Device].include?(params[:rootType])
               rm_users = (params[:rm_users] || []).map(&:to_i)
               obj.users.delete(User.where(id: rm_users)) if %w[Person].include?(params[:actionType])
-              obj.admins.delete(User.where(id: rm_users)) if %w[Group].include?(params[:rootType]) && %w[Person].include?(params[:actionType])
+              if %w[Group].include?(params[:rootType]) && %w[Person].include?(params[:actionType])
```

Comment:
```suggestion
              if params[:rootType] == 'Group' && params[:actionType] == 'Person'
```
anstelle der ganzen includes auf ein Array element

Reply:
Ja, ist besser